NuRadioMC/utilities/Veff.py

```diff
@@ -301,155 +323,160 @@ def get_Veff_Aeff_single(filename, trigger_names, trigger_names_dict, trigger_co
-                        usgids, comm1, comm2 = np.intersect1d(usgids, ugids_triggered, assume_unique=True, return_indices=True)  # select only the gids that triggered
-                        common_mask = np.isin(sgids, usgids)
-                        sgids = sgids[common_mask]  # also reduce gids array to the event groups that triggered
-                        if(len(usgids) == 0):  # skip stations that don't have any trigger for this trigger combination
-                            continue
-                        usgids_index = sorter[np.searchsorted(ugids_triggered, usgids, sorter=sorter)]
-                        # each station might have multiple triggeres per event group id. We need to select the one
-                        # event with the largest amplitude. Let's first check if one event group created more than one event
-                        max_amps_per_event_channel = np.nan_to_num(np.array(fin[key]['maximum_amplitudes_envelope'])[common_mask])
-                        max_amps_per_event = np.amax(max_amps_per_event_channel[:, channel_ids], axis=1)  # select the maximum amplitude of all considered channels
-                        if(len(sgids) != len(usgids)):
-                            # at least one event group created more than one event. Let's calculate it the slow but correct way
-                            for sgid in np.unique(sgids):  # loop over all event groups which triggered this station
-                                if(sgid not in usgids):
-                                    continue
-                                mask_gid = sgid == sgids  # select all event that are part of this event group
-                                index = np.squeeze(np.argwhere(ugids_triggered == sgid))
-                                max_amplitudes[index] = max(max_amplitudes[index], max_amps_per_event[mask_gid].max())
-                        else:
-                            max_amplitudes[usgids_index] = np.maximum(max_amplitudes[usgids_index], max_amps_per_event)
-                if('scale' in values['efficiency']):
-                    max_amplitudes *= values['efficiency']['scale']
-                if("Vrms" in values['efficiency']):
-                    Vrms = values['efficiency']['Vrms']
-                e = get_efficiency(max_amplitudes / Vrms)  # we calculated the maximum amplitudes for all gids, now we select only those that triggered
-                Veff = volume_proj_area * np.sum(weights[triggered] * e) / n_events
-                Vefferror = 0
-                if(np.sum(weights[triggered]) > 0):
-                    Vefferror = Veff / np.sum(weights[triggered] * e) ** 0.5
-                FC_low, FC_high = FC_limits(np.sum(weights[triggered] * e))
-                Veff_low = volume_proj_area * FC_low / n_events
-                Veff_high = volume_proj_area * FC_high / n_events
-
-            out[veff_aeff][trigger_name] = [Veff, Vefferror, np.sum(weights[triggered]), Veff_low, Veff_high]
```

Comment:
Was this line correct? The 3 item in the list should have been `np.sum(weights[triggered]) * e` correct? I fixed that

Reply:
Hi @cg-laser, I think this line 448 had a bug: For the calculation of `Veff_low` and `Veff_high` a `count` of `np.sum(weights[triggered] * e` (L 444) was used. However in the output only `np.sum(weights[triggered])` is stored for the count

Reply:
ah ok, got it, this was in the "efficiency" if statement. I implemented this ages ago and I'm not sure if anyone has been using this functionality in the last years ;-) But good to have this consistent now :-)